android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -411,23 +414,30 @@ constructor(
   private suspend fun Resource.applyRelatedEntityLocationMetaTag(
     questionnaireConfig: QuestionnaireConfig,
     context: Context,
+    subjectType: ResourceType?,
   ) {
-    questionnaireConfig.groupResource?.let {
-      if (it.groupIdentifier.isNotEmpty() && !it.removeGroup && !it.removeMember) {
-        val group =
-          loadResource(
-            ResourceType.Group,
-            it.groupIdentifier.extractLogicalIdUuid(),
-          )
-            as Group?
-        if (group != null) {
-          val system =
-            context.getString(
-              org.smartregister.fhircore.engine.R.string
-                .sync_strategy_related_entity_location_system,
-            )
-          group.meta.tag.filter { coding -> coding.system == system }.forEach(this.meta::addTag)
+    val resourceIdPair =
+      when {
+        !questionnaireConfig.resourceIdentifier.isNullOrEmpty() && subjectType != null -> {
+          Pair(subjectType, questionnaireConfig.resourceIdentifier!!)
+        }
+        !questionnaireConfig.groupResource?.groupIdentifier.isNullOrEmpty() &&
+          questionnaireConfig.groupResource?.removeGroup != true &&
+          questionnaireConfig.groupResource?.removeMember != true -> {
+          Pair(ResourceType.Group, questionnaireConfig.groupResource!!.groupIdentifier)
         }
+        else -> null
+      }
+    if (resourceIdPair != null) {
+      val (resourceType, resourceId) = resourceIdPair
+      val group =
+        loadResource(resourceType = resourceType, resourceIdentifier = resourceId) as Group?
+      if (group != null) {
+        val system =
+          context.getString(
+            org.smartregister.fhircore.engine.R.string.sync_strategy_related_entity_location_system,
+          )
+        group.meta.tag.filter { coding -> coding.system == system }.forEach(this.meta::addTag)
       }
```

Comment:
@ellykits  how do we handle the situation where we do not want to propagate the Group REL (Related Entity location) to the patient resources?

Reply:
The Group's REL only propagates if the Questionnaire has a `groupResource` (should only happen when adding Group member) config. If we do not want to propagate the group's REL two options:

1. Do not configure `groupResource` on resource 
2. Provide configurations for extracting location ID from QuestionnaireResponse when launching the questionnaire. The location ID will be used on the subject and related resources extracted from the questionnaire submission. (To be used when you want the configured locationId to override the subject's)

The subject resource's REL is given precedence over the Group's REL because, if the subject were added as a member of the Group it would have the Group's REL anyway.

Reply:
@ellykits  I assume point 2 is going to handle the situation where a Familly member does not leave in the same place as the rest of the family, right ?

Reply:
Yes that's correct.